cairo/ethereum/cancun/state.cairo

```diff
@@ -210,6 +211,68 @@ func get_account_optional{poseidon_ptr: PoseidonBuiltin*, state: State}(
     return account;
 }
 
+// @notice Returns the code for the given account.
+// If the code is not cached, it is loaded from the program input. The program
+// input must contain a Dict[Tuple[Low, High], Bytes] where the key is the codehash
+// (in its cairo representation) and the value is the code.
+// @dev: Accesses to the `code` field __MUST__ be done using `get_account_code`.
+func get_account_code{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    state: State,
+}(address: Address, account: Account) -> Bytes {
+    alloc_locals;
+
+    // Account code is already cached - return it
+    if (cast(account.value.code.value, felt) != 0) {
+        let res = Bytes(account.value.code.value);
+        return res;
+    }
+
+    // Account code is not cached - load it and ensure hash(code) == code_hash
+    // TODO: this should not be triggered in tests - only in prod script.
+    // TODO: ensure this works with a test.
+    let (code_) = alloc();
+    tempvar code = code_;
+    tempvar code_len: felt;
+    %{
+        # TODO: do we want to serialize (I fear it might be slow) here?
+        # best would be for the program input to already have the key:Tuple[Low, High] -> code
+        account_code = program_input["codehash_to_code"][(ids.account.value.code_hash.value.low, ids.account.value.code_hash.value.high)];
```

Comment:
why are we using a program_input in the inner codebase state.cairo?

might be a bigger / more tedious refactor but the program_input should be used only once to populated variables that are passed as arguments later no?

e.g. a last field in the State object codehash_to_code

Reply:
could be done in a subsequent issue